### PR TITLE
Challenges cyclic import problem fix

### PR DESCRIPTION
**Pull request by TroJan**
This pr updates the current code base, so that run server can be properly run. It comments out some import to resolve circular imports.

### CODE REVIEW

**Comment by dexter1691** [comment]
@TroJan: Thanks for pointing this out! 

Thanks for the cyclic import issue fix. 

The `challenge` field in the `ChallengeHostTeams` model is redundant since we already have that information in the form of `creator` field inside `ChallengeHostTeams`. This also makes sense since the same team can create multiple competitions. 

For instance: VQA team might create VQA Real Image Competition and VQA Abstract Image Competition. 

@deshraj: This also sounds like a good workflow. To create a competition, first create a team. We allow single user in the teams so that's alright. A team has to exist for a competition to be created.

File: apps/hosts/models.py [hunk 1]
```diff
@@ -21,17 +21,20 @@ class HostPermission(TimeStampedModel):
...
+    name = models.CharField(max_length=30)
     status = models.CharField(max_length=30, unique=True)
 
     def __unicode__(self):
         return self.name
 
 
 class ChallengeHostTeams(TimeStampedModel):
-    challenge = models.ForeignKey(Challenge)
+    # challenge = models.ForeignKey(Challenge)
```
Comment (dexter1691) [inline apps/hosts/models.py]: Can you completely get rid of this line.

File: apps/hosts/models.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 from __future__ import unicode_literals
 
 from accounts.models import (TimeStampedModel, UserStatus)
-from challenges.models import (Challenge)
+# from challenges.models import (Challenge)
```
Comment (dexter1691) [inline apps/hosts/models.py]: Same here. Please get rid of this.

File: settings/common.py [hunk 3]
```diff
@@ -11,10 +11,12 @@
 """
 
 import os
-
+import sys
 # Build paths inside the project like this: os.path.join(BASE_DIR, ...)
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+APPS_DIR = os.path.join(BASE_DIR, 'apps')
 
+sys.path.append(APPS_DIR)
```
Comment (deshraj) [inline settings/common.py]: I don't think we would need this since `apps` directory is already there on PYTHONPATH(root of the project).

**Comment by deshraj** [comment]
@dexter1691 This workflow sounds good to me. Also, it makes us a different from others. :-)

File: apps/hosts/models.py (hunk 1 above)
Comment (TroJan) [inline apps/hosts/models.py]: Sure

File: settings/common.py (hunk 3 above)
Comment (TroJan) [inline settings/common.py]: If i remove apps from system path, and change all import to be of the form `from apps.X.models import Y` I still get an error 'No module named X.models`

Any suggestions to get rid of this?

File: settings/common.py [hunk 4]
```diff
@@ -41,13 +43,13 @@
...
 
 OUR_APPS = [
-    'apps.analytics',
-    'apps.challenges',
-    'apps.hosts',
-    'apps.jobs',
-    'apps.participants',
-    'apps.teams',
-    'apps.web',
+    'analytics',
```
Comment (deshraj) [inline settings/common.py]: @TroJan can you add `accounts` here so that we can merge the PR?

**Pull request merged**